src/components/Main/Main.js

```diff
@@ -0,0 +1,34 @@
+import { useCountries } from '../../hooks/useCountries.js';
+import Country from '../Country/Country.js';
+import { useState } from 'react';
+import Controls from '../Controls/Controls.js';
+import './Main.css';
+
+export default function Main() {
+  const { countries, error, loading } = useCountries();
+  const [continent, setContinent] = useState('all');
+
+  const filtered = countries.filter(
+    (country) => country.continent === continent || continent === 'all'
+  );
+
+  if (loading && !error) {
```

Comment:
nice job using loading state

Reply:
also did we go over this or was this from demo code? i didn't think we'd covered this yet

Reply:
This was a combo of googling and asking classmates